lib/chef/knife/ssh.rb

```diff
@@ -160,6 +160,31 @@ def configure_session
         session_from_list(list)
       end
 
+      def get_ssh_attribute(node)
+        # Order of precedence for ssh target
+        # 1) command line attribute
+        # 2) configuration file
+        # 3) cloud attribute
+        # 4) fqdn
+        if config[:attribute_from_cli]
```

Comment:
Interesting, I should've checked where it is set.
I'll double check everything and update the PR as needed.